lib/stats.js

```diff
@@ -41,16 +42,32 @@ class Stats {
         }
     }
 
+    _Custom_requestContinued(fulfill, reject, params) {
+        const {timestamp, requestId} = params;
+        if (this.firstRequestId === requestId) {
+            this.requestContinuedMs = timestamp * 1000;
+        }
+        const entry = this.entries.get(requestId);
+        if (!entry) {
+            return;
+        }
+        entry.requestContinuedS = timestamp;
+    }
+
     _Page_domContentEventFired(fulfill, reject, params) {
         const {timestamp} = params;
-        this.domContentEventFiredMs = timestamp * 1000;
+        if (!this.domContentEventFiredMs) {
```

Comment:
Most changes are clear to me, can you help me understand this file changes better. 
Why do we set this values only if they have not been set. 
Is this for cases where we fall in here for a request which is not the first one?

Reply:
Yes.  It's to block out subsequent `Page.domContentEventFired` events.   We capture HAR logs for various scopes in the engine.  In a journey we may have multiple pages within our HAR.  However, the chrome-har-capturer is designed to only create a HAR for one page at a time.   

Page events in the engine are fired without request IDs to link them back to appropriate pages.  We only get a timestamp.  The Chrome engine does its best to group them to the appropriate page, but then everything gets grouped into a single array of logs by timestamp before sending to CHC.  So when it all comes to the chrome-har capturer it is creating a HAR for only a single page.  

The chrome-har-capturer blocks out subsequent page requests from being considered as pages and instead treats them as background requests.  But when any page events come in (load or DOM), they are currently being captured and overwriting the previous one.  So if we have a page that loads, then another page (or a reload) later had page events, they overwrite the original events and make the page load time seem really long.  So this change makes it so that only the first events (which are from the original document request) are considered.